staging/src/k8s.io/apiserver/pkg/registry/rest/resttest/resttest.go

```diff
@@ -910,45 +910,6 @@ func (t *Tester) testDeleteWithUID(obj runtime.Object, createFn CreateFunc, getF
 	}
 }
 
```

Comment:
@roycaihw Or maybe this should have been used in `TestDelete`, wdyt?

Reply:
Yes. Reading https://github.com/kubernetes/kubernetes/pull/74040 I think it should have been used in `TestDelete` :(

Could you call it twice with `dryRunOpts` and `opts` (like `testDeleteWithUID`) and see what happens? Thanks

Reply:
Works now.

The test failed initially because setting the resourceVersion does not work. So I have to retrieve the created object and use the resourceVersion from there.

I'm not sure if it's intended but the server did not reject the `resourceVersion`. According to:
https://github.com/kubernetes/kubernetes/blob/8b98305858b107369f2c9b9fd8ef1c5b0da078c0/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go#L144-L146
and 
https://github.com/kubernetes/kubernetes/blob/4c99949ae6b4a74e4ffda0caba86c151c23d579e/staging/src/k8s.io/apiserver/pkg/storage/etcd3/api_object_versioner.go#L84

It only returns `return errors.New("resourceVersion should not be set on objects to be created")` if it the resourceVersion can be parsed as `unit64`. So you can set whatever you want as `resourceVersion` as long as it's not a `unit64` and it won't complain (but ignore it anyway).

Reply:
I would expect the server to reject a create request if the RV is not empty. OTOH I think ignoring the non-uint64 RV is fine-- we still don't allow setting RV on create, at worst it causes confusion when someone tries to do it.

Reply:
Yup. It's also probably something we shouldn't change. Hard to say who somehow depends on it